src/lib/code-sample-editor.ts

```diff
@@ -1,72 +1,90 @@
-import { LitElement, html, customElement, css, property, TemplateResult, query, queryAll } from 'lit-element';
+import {
+  css,
+  customElement,
+  html,
+  LitElement,
+  property,
+  query,
+  queryAll,
+  TemplateResult
+} from 'lit-element';
 import { until } from 'lit-html/directives/until';
-import { FileRecord, CodeEditorTextarea } from './types';
-import { EMPTY_INDEX } from './constants';
-import { endWithSlash, generateUniqueSessionId, fetchProject, addFileRecordFromName, setUpServiceWorker } from './util';
-import * as Comlink from 'comlink';
-
 import './code-sample-editor-layout';
-import { SwControllerAPI } from '../sw';
+import { exposeOnSwIframeWindow } from './comlink-utils';
+import { EMPTY_INDEX, IFRAME_MODES } from './constants';
+import {
+  ClientServerAPI,
+  CodeEditorTextarea,
+  FileRecord,
+  ServiceWorkerRecord
+} from './types';
+import {
+  addFileRecordFromName,
+  endWithSlash,
+  fetchProject,
+  generateUniqueSessionId,
+  reloadIframeInIframe,
+  responseInitFromExtension,
+  setIframeContentsFromFile,
+  setUpServiceWorker
+} from './util';
+
+const generateDocumentAPI = (instance: CodeSampleEditor) => {
+  return class CodeSampleEditorRemote extends ClientServerAPI {
+    static async getResponseInitFromFilename(
+      path: string
+    ): Promise<{ payload: string; init: ResponseInit }> {
+      const pathParts = path.split('.');
```

Comment:
```ts
const extension = (path) => {
  const match = path.match(/\.([^\.\/]+)$/);
  return match ? match[1] : '';
}(path);
```

Reply:
meh probably better if
```
const extension = (path.split('/').pop() || '').split('.').pop();
```

Reply:
my reasoning here was that it would be an unsupported file extension which returns a 400 error. This is because I block creating files in the editor without a supported extension. Perhaps I should rethink `responseInitFromExtension` or this model altogether

Reply:
Wait I have 50 other ways to golf this line of code!!!

Reply:
Reasoning makes sense.  This was not a big deal I just got stuck on it because it was a comment in my started-review-from-two-weeks-ago-that-I-forgot